File: src/swingjs/plaf/JSWindowUI.java
package swingjs.plaf;

import jsjava.awt.Dialog;
import jsjava.awt.Font;
import jsjava.awt.FontMetrics;
import jsjava.awt.Graphics2D;
import jsjava.awt.Insets;
import jsjava.awt.JSComponent;
import jsjava.awt.Toolkit;
import jsjava.awt.Window;
import jsjava.awt.image.BufferedImage;
import jsjava.awt.peer.WindowPeer;
import jsjavax.swing.JComponent;
import jsjavax.swing.JWindow;
import swingjs.JSAppletViewer;
import swingjs.JSToolkit;
import swingjs.api.DOMNode;
import swingjs.api.HTML5Applet;

public class JSWindowUI extends JSComponentUI implements WindowPeer {

	protected DOMNode  
	/**************/ frameNode, /*********************/
  /************/ titleBarNode, /********************/
  /**/ titleNode,                      closerNode, //
  /***************/ layerNode, /********************/
  /*************/ menuBarNode; /********************/
  
	protected JWindow w;
	protected int z;

  protected int defaultWidth = 400;
  protected int defaultHeight = 400;
	

	protected boolean isFrame, isDialog;
	protected Window window;
	protected Font font;

	private Graphics2D graphics;

	/*
	 * Not Lightweight; an independent space with RootPane, LayeredPane,
	 * ContentPane, (optional) MenuBar, and GlassPane
	 * 
	 * 
	 * Used by JWindow, JFrame, JDialog, and JPopupMenu
	 * 
	 * 
	 * Lots to do here
	 * 
	 * @author Bob Hanson
	 */
	@Override
	public WindowPeer setFrame(Window target, boolean isFrame) {
		//set((JComponent)(Object)target); // yes, I know it is not a JComponent. This is JavaScript!
		window = target;
		w = (JWindow) window;
		this.isFrame = isFrame;
		isContainer = isWindow = true;
		JSComponent jc = (JSComponent) (Object) this;
		JSAppletViewer viewer = JSToolkit.getAppletViewer();
		applet = viewer.html5Applet;
		graphics = (Graphics2D) jc.getGraphics();
		return this;
	}

	@Override
	public DOMNode createDOMNode() {
		if (domNode == null) {
			containerNode = domNode = createDOMObject("div", id);
		}
		return domNode;
	}
	
	@Override
	protected DOMNode setHTMLElement() {
		
		DOMNode node = setHTMLElementCUI();
		return node;		
	}
	

	@Override
	public Toolkit getToolkit() {
		return Toolkit.getDefaultToolkit();
	}

	@Override
	public FontMetrics getFontMetrics(Font font) {
		if (!font.equals(this.font))
			this.window.setFont(this.font = font);
		return graphics.getFontMetrics(font);
	}


	@Override
	public void toFront() {
		System.out.println("window to front for " + jc.htmlName);
		// TODO Auto-generated method stub
		
	}

	@Override
	public void toBack() {
		System.out.println("window to back for " + jc.htmlName);
		// TODO Auto-generated method stub
		
	}

	@Override
	public void updateAlwaysOnTopState() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void updateFocusableWindowState() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public boolean requestWindowFocus() {
		// TODO Auto-generated method stub
		return false;
	}

	@Override
	public void setModalBlocked(Dialog blocker, boolean blocked) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void updateMinimumSize() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void updateIconImages() {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void setOpacity(float opacity) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void setOpaque(boolean isOpaque) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void updateWindow(BufferedImage backBuffer) {
		// TODO Auto-generated method stub
		
	}

	@Override
	public void repositionSecurityWarning() {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void installJSUI() {
		// TODO Auto-generated method stub
		
	}

	@Override
	protected void uninstallJSUI() {
	}

	@Override
	public void dispose() {
		JSToolkit.J2S._jsUnsetMouse(domNode);
		DOMNode.remove(outerNode);
	}

	@Override
	public Insets getInsets() {
		return new Insets(0, 0, 0, 0);
	}


}

File: site/swingjs/j2s/swingjs/plaf/JSWindowUI.js
Clazz.declarePackage ("swingjs.plaf");
Clazz.load (["java.awt.peer.WindowPeer", "swingjs.plaf.JSComponentUI"], "swingjs.plaf.JSWindowUI", ["java.lang.Thread", "java.awt.GraphicsEnvironment", "$.Insets", "$.Toolkit"], function () {
c$ = Clazz.decorateAsClass (function () {
this.frameZ = 0;
this.isFrame = false;
this.window = null;
this.applet = null;
this.graphics = null;
this.font = null;
Clazz.instantialize (this, arguments);
}, swingjs.plaf, "JSWindowUI", swingjs.plaf.JSComponentUI, java.awt.peer.WindowPeer);
Clazz.overrideMethod (c$, "setFrame", 
function (target, isFrame) {
this.frameZ = 19000;
this.isFrame = isFrame;
this.window = target;
var jc = this;
jc.myThread = Thread.currentThread ();
jc.threadGroup = jc.myThread.getThreadGroup ();
this.applet = (jc.threadGroup).getHtmlApplet ();
this.graphics = (java.awt.GraphicsEnvironment.getLocalGraphicsEnvironment ()).createGraphicsSized (target, 500, 300);
return this;
}, "java.awt.Window,~B");
Clazz.overrideMethod (c$, "getDOMObject", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getGraphics", 
function () {
this.graphics.setFont (this.window.getFont ());
return this.graphics;
});
Clazz.overrideMethod (c$, "getToolkit", 
function () {
return java.awt.Toolkit.getDefaultToolkit ();
});
Clazz.overrideMethod (c$, "getFontMetrics", 
function (font) {
if (!font.equals (this.font)) this.window.setFont (this.font = font);
return this.graphics.getFontMetrics (font);
}, "java.awt.Font");
Clazz.overrideMethod (c$, "getInsets", 
function () {
return  new java.awt.Insets (30, 2, 2, 2);
});
Clazz.overrideMethod (c$, "beginValidate", 
function () {
});
Clazz.overrideMethod (c$, "endValidate", 
function () {
});
Clazz.overrideMethod (c$, "beginLayout", 
function () {
});
Clazz.overrideMethod (c$, "endLayout", 
function () {
});
Clazz.overrideMethod (c$, "toFront", 
function () {
System.out.println ("window to front for " + this.c.htmlName);
});
Clazz.overrideMethod (c$, "toBack", 
function () {
System.out.println ("window to back for " + this.c.htmlName);
});
Clazz.overrideMethod (c$, "updateAlwaysOnTopState", 
function () {
});
Clazz.overrideMethod (c$, "updateFocusableWindowState", 
function () {
});
Clazz.overrideMethod (c$, "requestWindowFocus", 
function () {
return false;
});
Clazz.overrideMethod (c$, "setModalBlocked", 
function (blocker, blocked) {
}, "java.awt.Dialog,~B");
Clazz.overrideMethod (c$, "updateMinimumSize", 
function () {
});
Clazz.overrideMethod (c$, "updateIconImages", 
function () {
});
Clazz.overrideMethod (c$, "setOpacity", 
function (opacity) {
}, "~N");
Clazz.overrideMethod (c$, "setOpaque", 
function (isOpaque) {
}, "~B");
Clazz.overrideMethod (c$, "updateWindow", 
function (backBuffer) {
}, "java.awt.image.BufferedImage");
Clazz.overrideMethod (c$, "repositionSecurityWarning", 
function () {
});
Clazz.overrideMethod (c$, "installJSUI", 
function () {
});
Clazz.overrideMethod (c$, "uninstallJSUI", 
function () {
});
});
